internal/zinc-core/src/main/java/xsbti/compile/TastyFiles.java

```diff
@@ -0,0 +1,25 @@
+package xsbti.compile;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Can be added to `IncOptions.auxiliaryClassFiles` so that the TASTy files are
+ * managed by the ClassFileManager.
+ */
+public class TastyFiles implements AuxiliaryClassFiles {
+    @Override
+    public Path[] associatedFiles(Path classFile) {
+        // not all classfiles have a corresponding TASTy file but we strongly assume that the
+        // ClassFileManager is forgiving on files that do not exist.
+        String absolutePath = classFile.toAbsolutePath().toString();
+        if (absolutePath.endsWith(".class")) {
+            String prefix = absolutePath.substring(0, absolutePath.length() - 6);
+            Path tastyFile = Paths.get(prefix + ".tasty");
```

Comment:
This implementation will only work for `Path`s that represent the real file system. It would be better, IMO, to only use methods of `Path` that work for arbitrary `Path`s. Perhaps the following:
```suggestion
        String fileNameString = classFile.getFileName().toString();
        if (fileNameString.endsWith(".class")) {
            String fileNamePrefix = fileNameString.substring(0, fileNameString.length() - 6);
            Path tastyFile = classFile.resolveSibling(fileNamePrefix + ".tasty");
```

Reply:
The same comment applies to the other implementations of `AuxiliaryClassFiles`, obviously.

Reply:
I have applied those changes and reduced the code duplication by introducing a `AuxiliaryClassFileExtension` parent class